packages/playground/src/api/spatial/types.ts

```diff
@@ -0,0 +1,139 @@
+import { CogniteExternalId, CogniteInternalId, Limit } from '@cognite/sdk-core';
+
+export type AttributeType =
+  | 'STRING'
+  | 'LONG'
+  | 'DOUBLE'
+  | 'BOOLEAN'
+  | 'TIMESTAMP'
+  | 'GEOMETRY'
+  | 'POINT'
+  | 'LINESTRING'
+  | 'POLYGON'
+  | 'MULTIPOINT'
+  | 'MULTILINESTRING'
+  | 'MULTIPOLYGON'
+  | 'GEOMETRYZ'
+  | 'POINTZ'
+  | 'LINESTRINGZ'
+  | 'POLYGONZ'
+  | 'MULTIPOINTZ'
+  | 'MULTILINESTRINGZ'
+  | 'MULTIPOLYGONZ'
+  | 'GEOMETRYCOLLECTION';
+
+export interface Attributes {
+  [attribute: string]: {
+    type: AttributeType;
+    description?: string;
+    srid?: number;
+    size?: number;
+    optional?: boolean;
+  };
+}
+
+export interface Spatial {
+  [id: string]: unknown;
+}
+
+export interface FeaturesCreateItem {
+  externalId: CogniteExternalId;
+  [attribute: string]:
+    | string
+    | number
+    | boolean
+    | Date
+    | { wkt: string }
+    | { geojson: object };
+}
+
+export interface Features extends FeaturesCreateItem {
+  createdTime: Date;
+  lastUpdatedTime: Date;
+}
+
+export interface FeatureTypes {
+  id: CogniteInternalId;
+  externalId: CogniteExternalId;
+  attributes: Attributes & {
+    _created_at: { type: 'LONG' };
+    _updated_at: { type: 'LONG' };
+    _external_id: { type: 'STRING'; size: 32 };
+  };
+  searchSpec?: {
+    [searchName: string]: {
+      attributes:
+        | Array<keyof FeatureTypesCreateItem['attributes']>
+        | ['_created_at']
+        | ['_updated_at']
+        | ['_external_id'];
+    };
+  };
+}
+
+export interface FeatureTypesCreateItem {
+  externalId: CogniteExternalId;
+  attributes: Attributes;
+  searchSpec?: {
+    [searchName: string]: {
+      attributes: Array<keyof FeatureTypesCreateItem['attributes']>;
+    };
+  };
+}
+
+export interface FeatureSearchFilter extends Limit {
+  filter: OrFilter & AndFilter & NotFilter & AllFilters;
+}
+
+interface AttributeValue {
+  attribute: string;
+  value: unknown;
+}
+
+interface AllFilters
```

Comment:
can we work with json definitions for filters? if we add a new filter on the backend now we would need to change JS SDK too. If it was json-based, nothing would change here.

Reply:
or maybe js map instead of json - I don't know what is better way to do that in JS (I think in Python we used dictionary, is that right @vincent-cognite ?)

Reply:
@pawelzwolenskicognite I don't understand what json based means ? 🤔 
Do you want it to be accepting of any new type of filter being added with the json format of attribute value pair ?

Reply:
json based: instead of using TC classes that you defined below, we use json or map (like here in python SDK: https://github.com/cognitedata/cognite-sdk-python-experimental/blob/13aff08271ff381ad91856a25e7f5afac53ff769/cognite/experimental/_api/geospatial.py#L241) - so user can send any json/map and we don't need to hardcode TS classes for new filters

(there is validation of filters on server side, so if json is wrong, user will get 400)

Reply:
to be consistent with Python SDK it would be nice to use Map<String, Any> as a general filter and do not have any custom classes

@vincent-cognite can you share your opinion on that? :)

Reply:
I can also add a generic Map<string, any> within filter and keep these existing ones as well.
This would allow user to have autocomplete with all possible existing options and in case someone wishes to add a custom filter, they won't face any error and anyways a 400 would save the day at the end.

@pawelzwolenskicognite @vincent-cognite

Reply:
I would rather replace all the filters with Map<String, Any> so SDK will not require to be updated each time we add new filter type support on backend

Reply:
it seems way safter (better dx) to list the filters here explicitly and update the sdk when the backend changes. 
how frequently would happen anyway?